src/stactools/viirs/stac.py

```diff
@@ -1,117 +1,90 @@
-    Extent,
-    Item,
-    MediaType,
-    Provider,
-    ProviderRole,
-    SpatialExtent,
-    TemporalExtent,
-)
+import pystac.utils
+import stactools.core.utils.antimeridian
+from pystac import Asset, Item
 from pystac.extensions.projection import ProjectionExtension
+from stactools.core.io import ReadHrefModifier
+from stactools.core.utils.antimeridian import Strategy
 
-logger = logging.getLogger(__name__)
-
-
-def create_collection() -> Collection:
-    """Create a STAC Collection
-
-    This function includes logic to extract all relevant metadata from
-    an asset describing the STAC collection and/or metadata coded into an
-    accompanying constants.py file.
-
-    See `Collection<https://pystac.readthedocs.io/en/latest/api.html#collection>`_.
-
-    Returns:
-        Collection: STAC Collection object
-    """
-    providers = [
-        Provider(
-            name="The OS Community",
-            roles=[ProviderRole.PRODUCER, ProviderRole.PROCESSOR, ProviderRole.HOST],
-            url="https://github.com/stac-utils/stactools",
-        )
-    ]
-
-    # Time must be in UTC
-    demo_time = datetime.now(tz=timezone.utc)
-
-    extent = Extent(
-        SpatialExtent([[-180.0, 90.0, 180.0, -90.0]]),
-        TemporalExtent([[demo_time, None]]),
-    )
+from stactools.viirs import constants
+from stactools.viirs.fragment import STACFragments
+from stactools.viirs.metadata import viirs_metadata
+from stactools.viirs.utils import add_extensions
 
-    collection = Collection(
-        id="my-collection-id",
-        title="A dummy STAC Collection",
-        description="Used for demonstration purposes",
-        license="CC-0",
-        providers=providers,
-        extent=extent,
-        catalog_type=CatalogType.RELATIVE_PUBLISHED,
-    )
-
-    return collection
-
-
-def create_item(asset_href: str) -> Item:
-    """Create a STAC Item
+logger = logging.getLogger(__name__)
 
-    This function should include logic to extract all relevant metadata from an
-    asset, metadata asset, and/or a constants.py file.
 
-    See `Item<https://pystac.readthedocs.io/en/latest/api.html#item>`_.
+def create_item(
+    h5_href: str,
+    cog_hrefs: Optional[List[str]] = None,
+    read_href_modifier: Optional[ReadHrefModifier] = None,
+    antimeridian_strategy: Strategy = Strategy.SPLIT,
+    densify_factor: Optional[int] = None,
+) -> Item:
+    """Creates a STAC Item from VIIRS data.
 
     Args:
-        asset_href (str): The HREF pointing to an asset associated with the item
+        h5_href (str): href to an H5 (HDF5) file
+        cog_hrefs (List[str]): Optional list of COG asset TIF hrefs
+        read_href_modifier (Callable[[str], str]): An optional function to
+            modify the href (e.g. to add a token to a url)
+        antimeridian_strategy (Strategy): Either split on -180 or normalize
+            geometries so all longitudes are either positive or negative.
+            Default is to split antimeridian geometries.
```

Comment:
Needs `densify_factor` in the docstring.

Reply:
Yep. Did some additional docstringing as well.